### ISSUE TITLE
Nullpointer exception for BlockStateMock#equals() method

### ISSUE THREAD

**Issue by Thorinwasher** [MockBukkit/MockBukkit#875]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest version of MockBukkit?

- [X] I am using the latest version of MockBukkit.

### Minecraft Version

1.20

### Describe the bug

Whenever you create a BlockStateMock without any specified location and then use the BlockStateMock#equals() the following stacktrace happens
```
java.lang.NullPointerException: Cannot invoke "org.bukkit.Location.getWorld()" because "this.location" is null
	at be.seeseemelk.mockbukkit.block.BlockMock.getWorld(BlockMock.java:195)
	at be.seeseemelk.mockbukkit.block.state.BlockStateMock.getWorld(BlockStateMock.java:187)
	at be.seeseemelk.mockbukkit.block.state.BlockStateMock.equals(BlockStateMock.java:370)
```
(I'm working on a pr at the moment, so the lines might be a bit wrong here)

### Reproducible Test

```java

	@Test
	void equals_differentMaterial()
	{
		BlockStateMock blockState1 = (BlockStateMock) new BlockMock(Material.STONE).getState();
		BlockState blockState2 = blockState1.getSnapshot();
		assertEquals(blockState1, blockState2);
		blockState2.setType(Material.AIR);
		assertNotEquals(blockState1, blockState2);
	}
```

### Anything else?

What I'm wondering is what behavior should be expected here? Should there be a nullpointer exception, as the location is null, or should the equals take the null locations into consideration and allow two different states be equal to each other if both of their locations are null?

**Comment by SplotyCode** [comment]
Just ran into the same issue. I believe the comparison is not required at all https://github.com/MockBukkit/MockBukkit/pull/878

**Comment by Thorinwasher** [comment]
Fixed by #878
